core/include/core/dataio.h

```diff
@@ -19,11 +20,11 @@
  *                  read until EOF.
  * @param  timeout  Timeout in seconds for socket connections.
  * @param  buffersize Advisory buffer size in bytes for aggregating reads
- * @return stream   The input stream configured by this function.
+ * @param  ext      Required file extension (excluding compression suffixes)
```

Comment:
The `ext` parameters to `g3_*stream_*_path` don't seem to do much; they only provide a very weak check that files being opened might be what users expect them to be? They add a lot of complexity to do this, though, and the resulting error message ("Invalid filename") is rather uninformative. Given that this check was not present before, was there any significant rate of user mistakes that it would help?

Reply:
`Filename 'somefile' does not have expected extension 'blah'` would at least help the user know why they're being stopped and maybe what to do about it.